lib/commands/ping.js

```diff
@@ -0,0 +1,43 @@
+'use strict'
+
+const npmConfig = require('../config.js')
+const libnpm = require('libnpm')
+const figgyPudding = require('figgy-pudding')
+const log = require('npmlog')
+
+module.exports = ping
+
+function ping (argv) {
+  const {json, registry, silent} = argv
+
+  const PingConfig = figgyPudding({
+    json: {
+      default: json
+    },
+    registry: {
+      default: registry
+    }
+  })
+
+  const opts = PingConfig(npmConfig())
```

Comment:
Try using this instead:
```suggestion
  const opts = PingConfig(npmConfig().concat(argv))
```